### PR TITLE
issue: 778287 Initialize struct mce_sys members in the constructor

### PR DESCRIPTION
**Pull request by OphirMunk**
Currently mce_sys members are initialized in function get_env_params() which is
called during library initialization main_init(). However, in some distros it might
be too late as mce_sys members are referenced before the call to get_env_params().

Signed-off-by: Ophir Munk <ophirmu@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/71/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/73/ for details (Mellanox internal link).

File: src/vma/util/sys_vars.cpp [hunk 1]
```diff
@@ -0,0 +1,959 @@
...
+#include "vma/sock/fd_collection.h"
+#include "vma/sock/sockinfo_tcp.h"
+#include "vma/sock/sockinfo_udp.h"
+#include "vma/iomux/io_mux_call.h"
+
+#include "vma/util/instrumentation.h"
+
+void check_netperf_flags();
+
+struct mce_sys_var & mce_sys = mce_sys_var::instance();
```
Comment (ashanably) [inline src/vma/util/sys_vars.cpp]: please remove mce_sys since it is not use any more.

File: src/vma/util/sys_vars.h [hunk 2]
```diff
@@ -392,14 +392,24 @@ struct mce_sys_var {
 	sysctl_reader_t & sysctl_reader;
 
 private:
+	void get_env_params();
```
Comment (ashanably) [inline src/vma/util/sys_vars.h]: Please change get_env_params() visibility to public function so it can be accessed from sock-redirect.cpp.

**Comment by ashanably** [comment]
Please remove get_env_params() from main.h.
Please replace get_env_params() with safe_mce_sys().get_env_params() in sock-redirect.cpp.

**Pull request merged**